test/hotspot/jtreg/containers/docker/TestJcmd.java

```diff
@@ -0,0 +1,194 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+/*
+ * @test
+ * @summary Test JCMD across container boundary. The JCMD runs on a host system,
+ *          while sending commands to a JVM that runs inside a container.
+ * @requires docker.support
+ * @requires vm.flagless
+ * @modules java.base/jdk.internal.misc
+ *          java.management
+ *          jdk.jartool/sun.tools.jar
+ * @library /test/lib
+ * @build EventGeneratorLoop
+ * @run driver TestJcmd
+ */
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import jdk.test.lib.Container;
+import jdk.test.lib.JDKToolFinder;
+import jdk.test.lib.Platform;
+import jdk.test.lib.Utils;
+import jdk.test.lib.containers.docker.Common;
+import jdk.test.lib.containers.docker.DockerRunOptions;
+import jdk.test.lib.containers.docker.DockerfileConfig;
+import jdk.test.lib.containers.docker.DockerTestUtils;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import jtreg.SkippedException;
+
+
+public class TestJcmd {
+    private static final String IMAGE_NAME = Common.imageName("jcmd");
+    private static final int TIME_TO_RUN_CONTAINER_PROCESS = (int) (10 * Utils.TIMEOUT_FACTOR); // seconds
+    private static final String CONTAINER_NAME = "test-container";
+
+
+    public static void main(String[] args) throws Exception {
+        DockerTestUtils.canTestDocker();
+
+        // See JDK-8273216 for details
+        if (Container.ENGINE_COMMAND.equals("podman")) {
+            throw new SkippedException("JCMD does not work across container boundaries when using Podman");
+        }
+
+        // Need to create a custom dockerfile where user name and id, as well as group name and id
+        // of the JVM running in container must match the ones from the inspecting JCMD process.
+        String uid = getId("-u");
+        String gid = getId("-g");
+        String userName = getId("-un");
+        String groupName = getId("-gn");
+        String content = generateCustomDockerfile(uid, gid, userName, groupName);
+        DockerTestUtils.buildJdkContainerImage(IMAGE_NAME, content);
+
+        try {
+            Process p = startObservedContainer();
+            long pid = testJcmdGetPid("EventGeneratorLoop");
+
+            assertIsAlive(p);
+            testJcmdHelp(pid);
+
+            assertIsAlive(p);
+            testVmInfo(pid);
+
+            p.waitFor();
+        } finally {
+            DockerTestUtils.removeDockerImage(IMAGE_NAME);
+        }
+    }
+
+
+    // Run "jcmd -l" in a sidecar container, find the target process.
```

Comment:
Since 'jcmd -l' is not run in a sidecar, this comment is incorrect.

Reply:
Good catch. I will update the comment.